<launch>

  
  <arg name="input_rosbag_path" default="/home/box_ws/src/grand_tour_box/box_drivers/novatel_oem7_driver/src/novatel_oem7_driver/data/compare_hesai/combined2/withcpt/merge/merged_all.bag"/>
  
  <param name="/use_sim_time" value="true"/>
  <arg name="node_start_delay" default="2.0" />

  <!-- <node pkg="hesai_ros_driver" name="hesai_ros_driver_node" type="hesai_ros_driver_node" output="screen">
    <param name="config_path" type="string" value= "$(find hesai_ros_driver)/config/packet_replay.yaml"/>
    <param name="/use_sim_time" value="true"/>
  </node> -->

  <!-- rviz -->
  <!-- [ x: 0, y: -1.5707963, z: 1.5707963 ] -->
   <!-- <node pkg="tf" type="static_transform_publisher" name="box_to_robot" args="0.2178 0.076 0.0036 0 -1.5707963 1.5707963 base box_base 10" /> -->
 

  <node pkg="rviz" name="rviz" type="rviz" args="-d /home/box_ws/src/grand_tour_box/box_post_processing/rviz/rviz.rviz" required="true"/>

  <node pkg="rosbag" type="play" name="player" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' xterm -bg lightblue -fg darkslategrey -bd plum -bw 10 -geometry 80x40-50+250 -fn -sony-fixed-medium-r-normal--24-170-100-100-c-120-iso8859-1 -e " args="--clock -r 2 $(arg input_rosbag_path)"/>

  <!-- TF for Gps Frame -->
  <!-- x y z yaw pitch roll frame_id child_frame_id period_in_ms -->
  <!-- x y z qx qy qz qw frame_id child_frame_id  period_in_ms-->
  <!-- [ x: -1.5707963, y: 0, z: -1.5707963 ] -->
  <!-- { 0.5, -0.5, 0.5, -0.5 -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="box_to_robot" args="-0.0361 0.2178 0.07640 -1.5707963 0 1.5707963 base box_base 10" /> -->



  <rosparam command="load" file="/home/box_ws/src/grand_tour_box/box_drivers/novatel_oem7_driver/src/novatel_oem7_driver/data/compare_hesai/combined2/tf_sandbox/2024-08-14-10-45-39_lpc.yaml"/>
  <!-- Needs xterm -->
  <!-- sudo apt install xterm -->

  <!-- rosbag -->
  <!-- <node pkg="rosbag" type="play" name="player" output="screen" launch-prefix="xterm -bg lightblue -fg darkslategrey -bd plum -bw 10 -geometry 80x40-50+250 -e " args="clock -r 10 $(arg input_rosbag_path)"/> -->

</launch>

